### PR TITLE
Refactor how we get incidents

### PR DESCRIPTION
**Pull request by GrahamCampbell**
This will need testing by someone.

### CODE REVIEW

**Comment by GrahamCampbell** [comment]
// cc @joecohens, @jbrooksuk

**Comment by joecohens** [comment]
I'll do it

**Comment by joecohens** [comment]
Yessss!! This had to be done!

File: src/Controllers/HomeController.php [hunk 1]
```diff
@@ -19,10 +21,22 @@ public function showIndex()
     {
         $components = Component::orderBy('order')->orderBy('created_at')->get();
 
+        $allIncidents = [];
+
+        foreach (range(1, 8) as $i) {
+            $date = Carbon::now()->subDays($i);
```
Comment (joecohens) [inline src/Controllers/HomeController.php]: Should be called $incidentDate, or the other way around.

Comment (GrahamCampbell) [inline src/Controllers/HomeController.php]: No, this should be `date`.

Comment (joecohens) [inline src/Controllers/HomeController.php]: What I mean is that I'm getting a "Undefined variable: incidentDate" so either change the value of $incidentDate to $date or change the $incidentDate below to $date :)

**Comment by GrahamCampbell** [comment]
How does this look now?

**Comment by jbrooksuk** [comment]
Good to me, although untested here - on my phone.

**Comment by joecohens** [comment]
@GrahamCampbell Same issue

File: src/Controllers/HomeController.php [hunk 2]
```diff
@@ -19,10 +21,22 @@ public function showIndex()
...
+            ])->orderBy('created_at', 'desc')->get();
+            $allIncidents[] = ['date' => $date->format('jS F Y'), 'incidents' => $incidents];
+        }
+
         return View::make('index', [
-            'components' => $components,
-            'pageTitle'  => Setting::get('app_name'),
-            'aboutApp'   => Markdown::render(Setting::get('app_about')),
+            'components'    => $components,
+            'iallIncidents' => $allIncidents,
```
Comment (GrahamCampbell) [inline src/Controllers/HomeController.php]: typo here...

File: src/Controllers/HomeController.php [hunk 3]
```diff
@@ -19,10 +21,22 @@ public function showIndex()
     {
         $components = Component::orderBy('order')->orderBy('created_at')->get();
 
+        $allIncidents = [];
+
+        foreach (range(0, 7) as $i) {
+            $date = Carbon::now()->subDays($i);
+            $incidents = Incident::whereBetween('created_at', [
+                $incidentDate->format('Y-m-d').' 00:00:00',
```
Comment (joecohens) [inline src/Controllers/HomeController.php]: $incidentDate not defined.

Comment (GrahamCampbell) [inline src/Controllers/HomeController.php]: oops

**Comment by GrahamCampbell** [comment]
Is this working now?

**Comment by joecohens** [comment]
Yes now is working!

**Comment by GrahamCampbell** [comment]
I'll fix the cs, and merge...

**Comment by joecohens** [comment]
Awesome :)

**Pull request merged**